src/components/AuthenticationContext.tsx

```diff
@@ -0,0 +1,37 @@
+import React, { useState, createContext, useEffect } from 'react';
+import { isAccessTokenValid } from '../util/authHelpers';
+
+type AuthContextType = {
+  authenticated: boolean;
+  login: () => void;
+  logout: () => void;
+  authContextLoaded: boolean;
+};
+
+export const AuthContext = createContext({} as AuthContextType);
```

Comment:
Føles litt dirty å assert'e et tomt objekt til å være `AuthContextType`.
Spesielt mtp at du exporter denne så kan det være litt uforventet at en type kan ha andre verdier enn det typen tilsier.

Stemmer for å enten endre typen til at feltene kan være undefined, eller ha en uniontype for contexten sånn at typesjekkeren redder oss og vi slipper å gjøre sjekker på innholdet.

```suggestion
export const AuthContext = createContext<AuthContextType | null>(null);
```

Reply:
Da blir det iallefall ekstra håndtering på vært sted man bruker konteksten. Da vil den jo enten være AuthContextType  eller null.

En løsning som ikke ser så bra ut er å definere objektet som:
`{authenticated: false, 
authContextLoaded: false, 
login: () => console.log(''), 
logout: () => console.log('')}`.
 console.log('') kan bytte ut med noe annet som generer en void, var bare det første jeg kom på